modules/cudacodec/src/video_decoder.cpp

```diff
@@ -127,15 +123,41 @@ void cv::cudacodec::detail::VideoDecoder::create(const FormatInfo& videoFormat)
         CV_Error(Error::StsUnsupportedFormat, "Video source is not supported by hardware video decoder");
         CV_LOG_ERROR(NULL, "Video source is not supported by hardware video decoder.");
     }
+
+    if (videoFormat.histogramEnabled){
+        if (decodeCaps.bIsHistogramSupported) {
+            if (decodeCaps.nCounterBitDepth != 32) {
+                std::ostringstream warning;
+                warning << "Luma histogram output disabled due to current device using " << decodeCaps.nCounterBitDepth << " bit bins.  Histogram output only supports 32 bit bins.";
+                CV_LOG_INFO(NULL, warning.str());
+                videoFormat.histogramEnabled = false;
+            }
+            else {
+                videoFormat.histogramEnabled = true;
+                videoFormat.nCounterBitDepth = decodeCaps.nCounterBitDepth;
+                videoFormat.nMaxHistogramBins = decodeCaps.nMaxHistogramBins;
+            }
+        }
+        else {
+            CV_LOG_INFO(NULL, "Luma histogram output is not supported for current codec and/or on current device.");
```

Comment:
I'm not seeing the log output I would expect when histogram output is requested but not available, should I be able to see it in the build bot output?